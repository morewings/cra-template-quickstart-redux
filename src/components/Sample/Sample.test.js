import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Sample from 'components/Sample/Sample';

const sample = jest.fn();
const defaultProps = {
  increment: 0,
  sample,
};

describe('src > components > Sample.js', () => {
  beforeEach(() => {
    sample.mockClear();
  });
  it('renders without crashing', () => {
    const wrapper = shallow(<Sample {...defaultProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
