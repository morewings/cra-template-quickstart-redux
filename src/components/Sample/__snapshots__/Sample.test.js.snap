// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src > components > Sample.js renders without crashing 1`] = `
<div>
  <div>
    Hello from sample component!
  </div>
  <button
    onClick={[MockFunction]}
    type="button"
  >
    Click me!
  </button>
  <div>
    You clicked: 
    0
     times.
  </div>
</div>
`;
